### PR TITLE
Respect project name in create-daml-app

### LINKED ISSUE #6681: `daml new my-project create-daml-app` doesn’t use the directory name as the DAML project name
When I create a DAML project with `daml new` with the skeleton or quickstart templates, it will set the project name in _daml.yaml_ to the same name as the directory name specified.

However, when I use `daml new my-project create-daml-app`, it sets the project name to "create-daml-app". I found this quite annoying.

It appears this is because it would need to use the same name in _ui/package.json_. I would hope that we can templatify this file too.

### PR DESCRIPTION
**Pull request by cocreature**
Now `daml new foobar create-daml-app` creates a project called
`foobar` replacing the name everywhere. The tests now run twice once
with the original project name since that’s what we use in the GSG and
once with a modified one. I guess you could argue that only running
the second should be enough so I’m open to removing the first one.

fixes #6681

changelog_begin

- [DAML Assistant] `daml new foobar create-daml-app` now properly
  respects the project name and creates a project called `foobar`
  instead of hardcoding the name fo `create-daml-app`.

changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/CreateDamlAppTests.hs [hunk 1]
```diff
@@ -9,58 +9,76 @@ import Data.Aeson.Extra.Merge
...
+    gettingStartedGuideTest projectName = testCaseSteps "Getting Started Guide" $ \step ->
       withTempDir $ \tmpDir -> do
         step "Create app from template"
         withCurrentDirectory tmpDir $ do
-          callCommandSilent "daml new create-daml-app create-daml-app"
-        let cdaDir = tmpDir </> "create-daml-app"
+          callCommandSilent $ "daml new " <> projectName <> " create-daml-app"
+        let cdaDir = tmpDir </> projectName
+
+        hPutStrLn stderr cdaDir
```
Comment (garyverhaegen-da) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/CreateDamlAppTests.hs:64]: Is that meant to stay in there?

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/CreateDamlAppTestsMain.hs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- Wrapper around the tests so we can change the arguments
```
Comment (garyverhaegen-da) [inline daml-assistant/integration-tests/src/DA/Daml/Assistant/CreateDamlAppTestsMain.hs:4]: Am I understanding this correctly that this enables caching of the compilation of the other main?

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/CreateDamlAppTests.hs (hunk 1 above)
Comment (cocreature) [reply daml-assistant/integration-tests/src/DA/Daml/Assistant/CreateDamlAppTests.hs]: Good catch, removed!

File: daml-assistant/integration-tests/src/DA/Daml/Assistant/CreateDamlAppTestsMain.hs (hunk 2 above)
Comment (cocreature) [reply daml-assistant/integration-tests/src/DA/Daml/Assistant/CreateDamlAppTestsMain.hs:4]: Exactly

**Pull request merged**